apps/meteor/app/api/server/v1/users.ts

```diff
@@ -1255,6 +1256,38 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'users.getNames',
+	{ authRequired: true, validateParams: isUsersGetNamesParamsGETProps },
+	{
+		async get() {
+			let users: Pick<IUser, '_id' | 'username' | 'name'>[] = [];
+
+			if ('userIds' in this.queryParams) {
+				users = await Users.findByIds(this.queryParams.userIds, {
+					projection: {
+						name: 1,
+						username: 1,
+					},
+				}).toArray();
+			}
+
+			if ('usernames' in this.queryParams) {
+				users = await Users.findByUsernames(this.queryParams.usernames, {
+					projection: {
+						name: 1,
+						username: 1,
+					},
+				}).toArray();
+			}
+
+			return API.v1.success({
+				users,
+			});
```

Comment:
```suggestion
			const options = {
				projection: {
					name: 1,
					username: 1,
				},
			};

			if ('userIds' in this.queryParams) {
				return API.v1.success({
					users: await Users.findByIds(this.queryParams.userIds, options).toArray(),
				});
			}

			return API.v1.success({
				users: await Users.findByUsernames(this.queryParams.usernames, options).toArray(),
			});
```
Looks fine, but we usually avoid using let